src/main/java/org/sonarlint/intellij/telemetry/SonarLintTelemetry.java

```diff
@@ -0,0 +1,143 @@
+import com.intellij.openapi.components.ApplicationComponent;
+import com.intellij.openapi.project.ProjectManager;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.NotNull;
+import org.sonarlint.intellij.SonarApplication;
+import org.sonarsource.sonarlint.core.WsHelperImpl;
+
+public class SonarLintTelemetry implements ApplicationComponent {
+  private static final String STORAGE_FILENAME = "sonarlint_usage.json";
+  private final TelemetryClient worker;
+  private TelemetryStorage storage;
+  private ScheduledFuture<?> scheduledFuture;
+  private boolean error = false;
+
+  public SonarLintTelemetry(SonarApplication application, ProjectManager projectManager) {
+    this.worker = new TelemetryClient(new WsHelperImpl(), application, projectManager);
+  }
+
+  private void loadStorage() {
+    Path filePath = getStorageFilePath();
+
+    if (Files.exists(filePath)) {
+      try {
+        storage = TelemetryStorage.load(filePath);
+      } catch (IOException e) {
+        // ignore, we will recreate
+      }
+    }
+
+    if (storage == null) {
+      storage = new TelemetryStorage();
+      try {
+        storage.save(filePath);
+      } catch (IOException e) {
+        // mark as errored so that no data is sent
+        error = true;
+      }
+    }
+  }
+
+  public boolean enabled() {
+    return storage.enabled();
+  }
+
+  public void setEnabled(boolean enabled) {
+    if (enabled) {
+      schedule();
+    } else {
+      cancelScheduler();
+    }
+    storage.setEnabled(enabled);
+    saveData();
+  }
+
+  @Override public void initComponent() {
+    loadStorage();
+
+    try {
+      if (storage.enabled()) {
+        schedule();
+      }
+    } catch (Exception e) {
+      // fail silently
+      e.printStackTrace();
+    }
+  }
+
+  private void schedule() {
+    if (scheduledFuture == null && !error) {
+      scheduledFuture = JobScheduler.getScheduler().scheduleWithFixedDelay(() -> worker.run(storage),
+        1, TimeUnit.DAYS.toMinutes(1), TimeUnit.MINUTES);
+    }
+  }
+
+  private void cancelScheduler() {
+    if (scheduledFuture != null) {
+      scheduledFuture.cancel(false);
+      scheduledFuture = null;
+    }
+  }
+
+  public void markUsage() {
+    LocalDate today = LocalDate.now();
+    LocalDate lastUsed = storage.lastUseDate();
+    if (lastUsed == null || !lastUsed.equals(today)) {
+      storage.setLastUseDate(today);
+      storage.setNumUseDays(storage.numUseDays() + 1);
+      saveData();
+    }
+  }
+
+  @Override public void disposeComponent() {
+    cancelScheduler();
+    saveData();
+  }
+
+  private void saveData() {
+    try {
+      storage.save(getStorageFilePath());
+    } catch (IOException e) {
+      // fail silently
+      e.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148)